docs(express.json): clarify usage and add route-specific limit example

kushal4345 opened the pull request:
### Summary
This PR updates the `express.json()` documentation to:
- Clarify behavior and options in a more concise way.
- Add an example showing how to set different `limit` values for specific routes (e.g., `/upload`).

### Changes
- Added a "Different limits for specific routes" section with a code example.
- Maintained existing style and formatting consistent with the Express docs.
- Verified that markdown tables and headings render correctly.

### Reason
Closes #1421 requests additional clarity on using `express.json()` with different route limits.  
This helps developers quickly understand how to override defaults for specific endpoints without affecting others.

### Notes
- No changes to Express codebase, only to documentation.
- Follows current contributing and formatting guidelines.

<!--
Thank you for your pull request. Please provide a description and 
note the Certificate of Origin below. 

-->

<!--
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

kushal4345 committed:
docs(express.json): add example for route-specific limits (#1421)

kushal4345 committed:
fix(docs): remove extra semicolons in express.json.md

ShubhamOulkar reviewed (changes requested):
I don't think changes are required. Property table explained `limit` option very well and clear. See our [`use middleware page`](https://expressjs.com/en/guide/using-middleware.html) since this example explains how to use `express.json()` middleware.

kushal4345 commented:
Thanks for the review! 

I understand that the property table explains the `limit` option, but I think this example could still add value for developers. The table shows the option in isolation, while this example demonstrates how to apply different limits for specific routes (e.g., `/upload`) without affecting the global middleware.  

In my experience, beginners often struggle to connect configuration options to real-world route usage, so I felt this would make the docs more practical and beginner-friendly.  

Of course, I’m happy to adjust the wording or placement if you think a shorter version would fit better. Please let me know your thoughts!

ShubhamOulkar commented:
@kushal4345, I think updating [`use middleware page`](https://expressjs.com/en/guide/using-middleware.html) would be better. This page includes a built-in middleware section at the bottom.

kushal4345 committed:
docs: include express.json example in using middleware page

kushal4345 commented:
Thanks for the suggestion 

i have Added a new section titled “Different Limits for Specific Routes” to the middleware page.
This section includes a clear code example demonstrating how to set different rate limits for individual routes.
The update aims to make the middleware page more informative and helpful for developers by providing practical usage guidance.

kushal4345 commented:
Hi @ShubhamOulkar, I saw you requested changes, but I couldn’t find specific comments in the Files changed tab. Could you clarify what needs to be updated? I’ll make the adjustments accordingly.

bjohansebas reviewed (changes requested):
Thanks for your contribution. I agree with @ShubhamOulkar; it makes more sense for this to be under the [use middleware page](https://expressjs.com/en/guide/using-middleware.html), not to add it in the API files, because each API describes its functionality and shouldn’t explain how to use it under different forms of middleware.

bjohansebas commented on _includes/api/en/4x/express.static.md:
From what I can see, you made the change in the wrong file, judging by the commit message.

kushal4345 committed:
add express limit example in using middleware

kushal4345 committed:
fix the issue

kushal4345 commented:
Hi  I’ve updated the using-middleware.md guide to include an example of applying different JSON body size limits for specific routes.

Added a code snippet showing how to set a custom limit (5mb) for /upload.
Ensured the default express.json() still applies for other routes.
Fixed formatting to properly close the markdown code block.

bjohansebas commented on zh-tw/guide/using-middleware.md:
In this file neither, it’s in this one.
https://github.com/expressjs/expressjs.com/blob/gh-pages/en/guide/using-middleware.md

kushal4345 committed:
fix issues

kushal4345 committed:
Merge branch 'gh-pages' into 1421-docs-clean

kushal4345 commented:
Thanks for the clarification  I’ve added the new example for different limits inside en/guide/using-middleware.md under the Built-in middleware section as suggested. Please let me know if you’d like me to adjust the placement further.

ShubhamOulkar commented on en/guide/using-middleware.md:
```diff
@@ -252,7 +252,20 @@ Express has the following built-in middleware functions:
 - [express.static](/en/5x/api.html#express.static) serves static assets such as HTML files, images, and so on.
 - [express.json](/en/5x/api.html#express.json) parses incoming requests with JSON payloads. **NOTE: Available with Express 4.16.0+**
 - [express.urlencoded](/en/5x/api.html#express.urlencoded) parses incoming requests with URL-encoded payloads.  **NOTE: Available with Express 4.16.0+**
+ gh-pages
```
```suggestion
```
cleanup

ShubhamOulkar commented on zh-tw/guide/using-middleware.md:
```diff
@@ -36,67 +36,6 @@ You can also load a series of middleware functions together, which creates a sub
 
 Bind application-level middleware to an instance of the [app object](/{{ page.lang }}/5x/api.html#app) by using the `app.use()` and `app.METHOD()` functions, where `METHOD` is the HTTP method of the request that the middleware function handles (such as GET, PUT, or POST) in lowercase.
 
-This example shows a middleware function with no mount path. The function is executed every time the app receives a request.
```
These changes are not required. Please undo it.

ShubhamOulkar commented on en/guide/using-middleware.md:
```diff
@@ -252,7 +252,20 @@ Express has the following built-in middleware functions:
 - [express.urlencoded](/en/5x/api.html#express.urlencoded) parses incoming requests with URL-encoded payloads.  **NOTE: Available with Express 4.16.0+**
+ gh-pages
 
+Here is an example of using the `express.json` middleware with a custom limit:
```
🤔 Should we need separate pages for explaining how to use built in middleware options?
It feels off-topic on the page

ShubhamOulkar reviewed (changes requested):
Please remove unnecessary changes

kushal4345 committed:
Update en/guide/using-middleware.md

Co-authored-by: shubham oulkar <91728992+ShubhamOulkar@users.noreply.github.com>

jonchurch commented:
The root issue https://github.com/expressjs/express/issues/3932 that inspired the linked issue is really about understanding middleware ordering and targeting specific routes with specific middleware, not about `bodyParser.limit` setting really.

I don't think the proposed change focused on `bodyParser.limit` is necessary

karlhorky commented:
@jonchurch hmm not sure what you mean.

I understood expressjs/express#3932 to be about "how can I configure the `/upload` route with a certain limit and non-`/upload` routes with a different limit?" since that is the case provided in that issue description at the top of the page.

I think my issue #1421 doesn't subvert that original wish, and that the changes in this PR are helpful for understanding how to achieve this common use case.

Do you disagree with either of those things?
